starcheck/src/starcheck.pl

```diff
@@ -50,107 +46,15 @@
-
-def starcheck_version():
-    return version
-
-def get_data_dir():
-    sc_data = os.path.join(os.path.dirname(starcheck.__file__), 'data')
-    return sc_data if os.path.exists(sc_data) else ""
-
-def _make_pcad_attitude_check_report(kwargs):
-    try:
-       return make_pcad_attitude_check_report(**de_bytestr(kwargs))
-    except Exception as err:
-       perl.warning("Error running dynamic attitude checks {}\n".format(err))
-
-
-def get_dither_kadi_state(date):
-    date = date.decode('ascii')
-    cols = ['dither', 'dither_ampl_pitch', 'dither_ampl_yaw', 'dither_period_pitch', 'dither_period_yaw']
-    state = states.get_continuity(date, cols)
-    # Cast the numpy floats as plain floats
-    for key in ['dither_ampl_pitch', 'dither_ampl_yaw', 'dither_period_pitch', 'dither_period_yaw']:
-        state[key] = float(state[key])
-    # get most recent change time
-    state['time'] = float(np.max([DateTime(state['__dates__'][key]).secs for key in cols]))
-    return state
-
-
-def get_run_start_time(run_start_time, backstop_start):
-    """
-    Determine a reasonable reference run start time based on the supplied
-    run start time and the time of the first backstop command.  This
-    code uses a small hack so that a negative number is interpreted
-    as the desired "days back" from the backstop start time.  All other
-    Chandra.Time compatible formats for run start are used as absolute
-    times (which will then be passed to the thermal model code as the
-    time before which telemetry should be found for an initial state).
-    Note that the logic to determine the initial state will not allow
-    that state to be after backstop start time.
-
-    :param run_start_time: supplied run start time in a Chandra.Time format,
-                      empty string interpreted as "now" as expected,
-                      negative numbers special cased to be interpreted as
-                      "days back" relative to first backstop command.
-    :param backstop_start: time of first backstop command
-    :returns: YYYY:DOY string of reference run start time
-    """
-
-    run_start_time = de_bytestr(run_start_time)
-    backstop_start = de_bytestr(backstop_start)
-
-    # For the special case where run_start_time casts as a float
-    # check to see if it is negative and if so, set the reference
-    # time to be a time run_start_time days back from backstop start
-    try:
-        run_start_time = float(run_start_time)
-    # Handle nominal errors if run_start_time None or non-float Chandra.Time OK string.
-    except (TypeError, ValueError):
-        ref_time = DateTime(run_start_time)
-    else:
-        if run_start_time < 0:
-            ref_time = DateTime(backstop_start) + run_start_time
-        else:
-            raise ValueError("Float run_start_time should be negative")
-    return ref_time.date
+from starcheck.pcad_att_check import check_characteristics_date
+from starcheck.utils import (_make_pcad_attitude_check_report,
```

Comment:
YES!!!!  Excellent idea.

Reply:
Well, it was your idea and it was a good one!  So much nicer to keep the Python in a separate file and just convenient for editing.  I think some of the exception handling and wrappers still need simplification and improvement, but moving them into the separate file is a good start.